drgn/helpers/linux/printk.py

```diff
@@ -259,21 +262,44 @@ def get_dmesg(prog: Program) -> bytes:
     If you need to format the log buffer differently, use
     :func:`get_printk_records()` and format it yourself.
     """
-    lines = [
-        b"[% 5d.%06d] %s"
-        % (
-            record.timestamp // 1000000000,
-            record.timestamp % 1000000000 // 1000,
-            record.text,
+    if human_readable_time:
+        boot_time_s = int(ktime_get_coarse_real_ns(prog)) - int(
+            ktime_get_coarse_ns(prog)
         )
-        for record in get_printk_records(prog)
-    ]
-    lines.append(b"")  # So we get a trailing newline.
+        lines = [
+            b"[%s] %s"
+            % (
+                datetime.fromtimestamp(
+                    (boot_time_s + record.timestamp) // 1000000000, timezone.utc
+                )
+                .strftime("%a %b %e %T %Z %Y")
+                .encode("utf-8"),
+                record.text,
+            )
+            for record in get_printk_records(prog)
+        ]
+        lines.append(b"")
+    else:
+        lines = [
+            b"[% 5d.%06d] %s"
+            % (
+                record.timestamp // 1000000000,
+                record.timestamp % 1000000000 // 1000,
+                record.text,
+            )
+            for record in get_printk_records(prog)
+        ]
+        lines.append(b"")  # So we get a trailing newline.
     return b"\n".join(lines)
 
 
 @takes_program_or_default
-def print_dmesg(prog: Program, *, file: "Optional[SupportsWrite[str]]" = None) -> None:
+def print_dmesg(
+    prog: Program,
+    human_readable_time: bool = False,
+    *,
```

Comment:
Same here, keyword-only:
```suggestion
    *,
    human_readable_time: bool = False,
```